INTEGRATION.md

```diff
@@ -0,0 +1,230 @@
+
+* [`/last_blocks` docs](src/gui/README.md#get-last-n-blocks)
+* [`/blocks` docs](src/gui/README.md#get-blocks-in-specific-range)
+* [`/outputs` docs](src/gui/README.md#get-unspent-output-set-of-address-or-hash)
+* [`/balance` docs](src/gui/README.md#get-balance-of-addresses)
+
+#### Using skycoin as a library in a Go application
+
+We recommend using the [Skycoin REST API Client](https://godoc.org/github.com/skycoin/skycoin/src/gui#Client).
+
+### Sending coins
+
+#### General principles
+
+After each spend, wait for the transaction to confirm before trying to spend again.
+
+For higher throughput, combine multiple spends into one transaction.
+
+Skycoin uses "coin hours" to ratelimit transactions.
+The total number of coinhours in a transaction's outputs must be 50% or less than the number of coinhours in a transaction's inputs,
+or else the transaction is invalid and will not be accepted. A transaction must have at least 1 input with at least 1 coin hour.
+Sending too many transactions in quick succession will use up all available coinhours.
+Coinhours are earned at a rate of 1 coinhour per coin per hour, calculated per second.
+This means that 3600 coins will earn 1 coinhour per second.
+However, coinhours are only updated when a new block is published to the blockchain.
+New blocks are published every 10 seconds, but only if there are pending transactions in the network.
+
+To avoid running out of coinhours in situations where the application may frequently send,
+the sender should batch sends into a single transaction and send them on a
+30 second to 1 minute interval.
+
+There are other strategies to minimize the likelihood of running out of coinhours, such
+as splitting up balances into many unspent outputs and having a large balance which generates
+coinhours quickly.
+
+#### Using the CLI
+
+When sending coins from the CLI tool, a wallet file local to the caller is used.
+The CLI tool allows you to specify the wallet file on disk to use for operations.
+
+See [CLI command API](cmd/cli/README.md) for documentation of the CLI interface.
+
+To perform a send, the preferred method follows these steps in a loop:
+
+* `skycoin-cli createRawTransaction -m '[{"addr:"$addr1,"coins:"$coins1"}, ...]` - `-m` flag is send-to-many
+* `skycoin-cli broadcastTransaction` - returns `txid`
+* `skycoin-cli transaction $txid` - repeat this command until `"status"` is `"confirmed"`
+
+That is, create a raw transaction, broadcast it, and wait for it to confirm.
+
+Alternatively, `skycoin-cli send -m` could be used, which will create a transaction and broadcast it in one step.
+However, suppose you are updating your local database at the same time;
+
+* If you update your database after the `send` and database transaction fails, you cannot undo the broadcast, and your database will be inconsistent.
+* If you update your database before the `send` and the `send` fails, you will
```

Comment:
is this incomplete?

Reply:
Yes, I forget what the exact problem with this was. Something with obtaining the txid. Will remove it.